[docs] Add missing `justifyContent` values and update box styling

omarmosid opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [X] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Closes: #29050

![image](https://user-images.githubusercontent.com/47219640/137638628-5d9e7124-662c-4981-9cad-1254bd3b4e7b.png)

siriwatknp commented:
@omarmosid Thanks for pointing that out. I think all of the boxes in grey can be converted to blue (similar to grid demos) in this PR! please go ahead with blue boxes.

BTW, Nice work!

omarmosid commented:
Thankyou for the feedback @siriwatknp.

I have now updated the PR and added a few missing examples for css properties that can be used for `flexDirection`, `alignItems`, `alignContent`, `alignSelf`, `display`, `flexDirection`, `flexGrow`, `flexShrink`, `flexWrap`, and `Order`

I have also taken the liberty to add links to the relevent MDN documentation under each property title. Hope that's going to be okay. It's worth noting here that these links open in the same page since they are your normal markdown links and we may want to have them open in a new tab. So I could change:

```md
For more information please see [flex-direction](https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction) on MDN.
```

with
```md
For more information please see 
<a href="https://developer.mozilla.org/en-US/docs/Web/CSS/flex-direction" target="_blank" rel="noopener noreferrer">flex-direction</a> 
on MDN.
```

siriwatknp commented:
@omarmosid If you don't mind, I would like to reduce the font-size from `19px` to `16px`.
